man/sd_bus_negotiate_fds.xml

```diff
@@ -45,69 +46,67 @@
     <citerefentry><refentrytitle>sd_bus_message_get_realtime_usec</refentrytitle><manvolnum>3</manvolnum></citerefentry>,
     <citerefentry><refentrytitle>sd_bus_message_get_seqnum</refentrytitle><manvolnum>3</manvolnum></citerefentry>
-    to query the timestamps of incoming messages. If negotiation is disabled or not supported, these
-    calls will fail with <constant>-ENODATA</constant>. Note that currently no transports support
-    timestamping of messages. By default, message timestamping is not negotiated for
-    connections.</para>
+    to query the timestamps of incoming messages. If negotiation is disabled or not supported, these calls
+    will fail with <constant>-ENODATA</constant>. Note that currently no transports support timestamping of
+    messages. By default, message timestamping is not negotiated for connections.</para>
 
     <para><function>sd_bus_negotiate_creds()</function> controls whether and which implicit sender
-    credentials shall be attached automatically to all incoming messages. Takes a bus object and a
-    boolean indicating whether to enable or disable the credential parts encoded in the bit mask
-    value argument. Note that not all transports support attaching sender credentials to messages,
-    or do not support all types of sender credential parameters, or might suppress them under
-    certain circumstances for individual messages. Specifically, dbus1 only supports
-    <constant>SD_BUS_CREDS_UNIQUE_NAME</constant>. The sender credentials are suitable for
-    authorization decisions. By default, only <constant>SD_BUS_CREDS_WELL_KNOWN_NAMES</constant> and
-    <constant>SD_BUS_CREDS_UNIQUE_NAME</constant> are enabled. In fact, these two credential fields
-    are always sent along and cannot be turned off.</para>
-
-    <para>The <function>sd_bus_negotiate_fds()</function> function may
-    be called only before the connection has been started with
+    credentials shall be attached automatically to all incoming messages. Takes a bus object and a boolean
+    indicating whether to enable or disable the credential parts encoded in the bit mask value argument. Note
+    that not all transports support attaching sender credentials to messages, or do not support all types of
+    sender credential parameters, or might suppress them under certain circumstances for individual messages.
+    Specifically, dbus1 only supports <constant>SD_BUS_CREDS_UNIQUE_NAME</constant>. The sender credentials
+    are suitable for authorization decisions. By default, only
+    <constant>SD_BUS_CREDS_WELL_KNOWN_NAMES</constant> and <constant>SD_BUS_CREDS_UNIQUE_NAME</constant> are
+    enabled. In fact, these two credential fields are always sent along and cannot be turned off.</para>
+
+    <para><function>sd_bus_get_creds_mask()</function> returns the complete set of sender credentials
+    attached to all incoming messages in <parameter>mask</parameter>.</para>
```

Comment:
creds_mask was primarily useful on kdbus, since there we could tweak what kinda of metadata we'd get attached, and this function would tell you what was actually settled on. I am still hoping we can get something similar on classic dbus eventually, hence it's probably nothing to get rid of.

So the original idea was to use sd_bus_negotiate_creds() to say which ones you want, and then check with sd_bus_get_creds_mask() the ones we actually could get. It's still an upper bound only, however, since there might be weird synthetic messages, where we can't get everything. hence maybe say:
 
```
<function>sd_bus_get_creds_mask()</function> returns the set of sender credentials that was negiotiated to be attached to all incoming messages in <parameter>mask</parameter>. This value is an upper boundary only, and it's always necessary to check explicitly which credentials are attached to a specific message before using it.
```

or so?